snakemake/sourcecache.py

```diff
@@ -126,13 +126,24 @@ def __init__(
         self.repo_path = repo_path
         self.path = path
 
+    def get_basedir(self):
+        root_path, file_path, ref = split_git_path(self.get_path_or_uri())
+        file_name = "{}/{}".format(root_path, file_path).replace(root_path, "")
+        return LocalGitFile("file:/{}".format(root_path), file_name, ref=ref)
```

Comment:
What's wrong with the get_basedir method of the SourceFile class?

Reply:
I have update the code